src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -1,952 +1,382 @@
-                    await _cacheService.DeleteItemAsync(new PrivateSubjectMetaCacheKey(
-                        releaseVersionId: releaseVersionId,
-                        subjectId: deletePlan.SubjectId));
-                })
-                .OnSuccessVoid(() => _releaseDataFileService.Delete(releaseVersionId, fileId));
-        }
+                rv.ApprovalStatus,
+                rv.Published
+            })
+            .ToListAsync(cancellationToken: cancellationToken);
 
-        public async Task<Either<ActionResult, DataImportStatusViewModel>> GetDataFileImportStatus(
-            Guid releaseVersionId,
-            Guid fileId)
-        {
-            return await _persistenceHelper
-                .CheckEntityExists<ReleaseVersion>(releaseVersionId)
-                .OnSuccess(_userService.CheckCanViewReleaseVersion)
-                .OnSuccess(async _ =>
-                {
-                    // Ensure file is linked to the Release by getting the ReleaseFile first
-                    var releaseFile =
-                        await _releaseFileRepository.Find(releaseVersionId: releaseVersionId, fileId: fileId);
-                    if (releaseFile == null || releaseFile.File.Type != FileType.Data)
-                    {
-                        return DataImportStatusViewModel.NotFound();
-                    }
-
-                    return await _dataImportService.GetImportStatus(fileId);
-                });
-        }
+        var existsUnpublishedApprovedReleaseVersions = releaseVersions
+            .Any(rv => rv.ApprovalStatus == ReleaseApprovalStatus.Approved && rv.Published is null);
 
-        private async Task<Either<ActionResult, ReleaseFile>> CheckReleaseDataFileExists(
-            Guid releaseVersionId, Guid fileId)
-        {
-            return await _context.ReleaseFiles
-                .Include(rf => rf.File)
-                .Where(rf => rf.ReleaseVersionId == releaseVersionId)
-                .Where(rf => rf.FileId == fileId)
-                .SingleOrNotFoundAsync()
-                .OnSuccess(rf => rf.File.Type != FileType.Data
-                    ? new Either<ActionResult, ReleaseFile>(ValidationActionResult(FileTypeMustBeData))
-                    : rf);
-        }
+        return existsUnpublishedApprovedReleaseVersions
+            ? ValidationActionResult(ReleaseUndergoingPublishing)
+            : Unit.Instance;
+    }
 
-        private async Task<Either<ActionResult, Unit>> ValidateReleaseSlugUniqueToPublication(
-            string slug,
-            Guid publicationId,
-            Guid? releaseId = null)
-        {
-            var slugAlreadyExists = await _context.Releases
-                .Where(r => r.PublicationId == publicationId)
-                .AnyAsync(r => r.Slug == slug && r.Id != releaseId);
+    private async Task<Either<ActionResult, Unit>> ValidateReleaseRedirectDoesNotExistForNewSlug(Guid releaseId, string newReleaseSlug, CancellationToken cancellationToken)
```

Comment:
I think it's worth checking the formatting in this entire class. There's a long line here.